cmd/processor/app/processor.go

```diff
@@ -649,3 +657,49 @@ func (p *Processor) setWorkersStatus(triggerInstance trigger.Trigger, status sta
 
 	return nil
 }
+
+// creates a signal handler, so on signal processor gracefully terminated
+func (p *Processor) handleSignals() {
+	var captureSignal = make(chan os.Signal, 1)
+	signal.Notify(captureSignal, syscall.SIGTERM, syscall.SIGABRT, syscall.SIGINT)
+	p.actOnSignal(<-captureSignal)
+}
+
+func (p *Processor) actOnSignal(signal os.Signal) {
+	p.logger.WarnWith("", "Got system signal", signal.String())
+	p.terminateAllTriggers()
```

Comment:
As we discussed, `actOnSignal` can be a more general function that acts on different signals with a `switch-case`.

If you want to only implement this termination for now, just pass the channel output to `terminateAllTriggers` and skip this middle-man function.